tests/cpp/test_matmul_aten_evaluation.cpp

```diff
@@ -393,4 +408,79 @@ TEST_F(MatmulATenEvaluationTest, LinearWithBias) {
 
   EXPECT_TRUE(at::allclose(out[0], out_ref));
 }
+
+TEST_P(ATenNodesParametrizedTest, MatmulNodeConcrete) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  const auto& [a_shape, b_shape] = GetParam();
+
+  auto tv0 = makeConcreteTensor(a_shape, DataType::Half);
+  auto tv1 = makeConcreteTensor(b_shape, DataType::Half);
+  auto tv2 = eagerMatmul(tv0, tv1);
+
+  fusion->addInput(tv0);
+  fusion->addInput(tv1);
+  fusion->addOutput(tv2);
+
+  at::Tensor t0 = at::randn(a_shape, at::kHalf).cuda();
+  at::Tensor t1 = at::randn(b_shape, at::kHalf).cuda();
+  at::Tensor out_ref = at::matmul(t0, t1);
+
+  FusionExecutor fe;
+  fusion->aliasOutputToInput(
+      fusion->outputs()[0], /*input=*/nullptr, AllocationType::Evaluate);
+  fe.compileFusion(fusion.get(), {t0, t1});
+  auto out = fe.runFusion({t0, t1});
+
+  // Verify that fusion compilation was skipped.
+  EXPECT_FALSE(fe.hasCompiledKernel());
+
+  EXPECT_TRUE(at::allclose(out[0], out_ref));
+}
+
+TEST_P(ATenNodesParametrizedTest, MatmulNodeSymbolic) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  const auto& [a_shape, b_shape] = GetParam();
+
+  auto tv0 = makeSymbolicTensor(a_shape.size(), DataType::Half);
+  auto tv1 = makeSymbolicTensor(b_shape.size(), DataType::Half);
```

Comment:
```suggestion
  auto tv0 = makeSymbolicTensor(a_shape, DataType::Half);
  auto tv1 = makeSymbolicTensor(b_shape, DataType::Half);
```
This will ensure the tensors are defined in the fusion almost the way they would be using `fd.from_pytorch` in case the shapes contain `1`s. That could be useful here since you might translate some ops to non-`MatmulOp` if they are trivial. To be more precise though, this will still not declare them as contiguous. For that you might want to do
```c++
  auto tv0 = TensorViewBuilder().shape(sym_shape).dtype(DataType::Half).contiguity(true).build();
```
where `sym_shape` is a vector of `1` and `-1`.

Reply:
```
To be more precise though, this will still not declare them as contiguous
```

Why do we need this?

Reply:
However if in the future we try to evaluate this test by forcing the nvfuser matmul scheduler we might use an inefficient kernel since we won't be able to tell the inputs are contiguous. It's not a high priority, but it would be nice to have a test utility that could take an `at::Tensor` and create a `TensorView*` that matches it just like how we do in `fd.from_pytorch`.